gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala

```diff
@@ -75,15 +75,24 @@ object TransformHints {
   }
 
   def tag(plan: SparkPlan, hint: TransformHint): Unit = {
-    if (isAlreadyTagged(plan)) {
-      if (isNotTransformable(plan) && hint.isInstanceOf[TRANSFORM_SUPPORTED]) {
-        throw new UnsupportedOperationException(
-          s"Plan was already tagged as non-transformable, " +
-            s"cannot mark it as transformable after that: ${plan.toString()}")
+    val mergedHint = getHintOption(plan)
+      .map {
+        case TRANSFORM_UNSUPPORTED(Some(originalReason)) =>
+          hint match {
+            case TRANSFORM_UNSUPPORTED(newReason) =>
```

Comment:
Can we distinguish the cases for `hint`? `TRANSFORM_UNSUPPORTED(Some(newReason))`
`TRANSFORM_UNSUPPORTED(None)`

For the 2nd, I think we should do nothing.

It may be logically clear if we separate the tagging impl. for supported and unsupported hint. Thanks!